app/src/test/java/com/segment/analytics/android/integrations/appsflyer/AppsflyerIntegration_ConversionListenerTests.java

```diff
@@ -0,0 +1,274 @@
+//                .putValue("ad_group", "");
+//        Properties properties = new Properties().putValue("provider", "AppsFlyer");
+//        properties.putAll(attributionData);
+//        properties.remove("media_source");
+//        properties.remove("adgroup");
+//        properties.putValue("campaign", campaign);
+//        verify(analytics,times(1)).track("Install Attributed", properties);
+//        reset(analytics);
+//    }
+
+    @Test
+    public void testAppsflyerIntegration_ConversionListener_getFlag_happyFlow() throws Exception {
+        String key="key";
+        Analytics analytics = mock(Analytics.class);
+        Application app = mock(Application.class);
+        Context context = mock(Context.class);
+        SharedPreferences sharedPreferences = mock(SharedPreferences.class);
+        when(analytics.getApplication()).thenReturn(app);
+        when(app.getApplicationContext()).thenReturn(context);
+        when(context.getSharedPreferences("appsflyer-segment-data",0)).thenReturn(sharedPreferences);
+        when(sharedPreferences.getBoolean(key,false)).thenReturn(true);
+        AppsflyerIntegration.ConversionListener conversionListener = spy(new AppsflyerIntegration.ConversionListener(analytics));
+        Method getFlagMethod = AppsflyerIntegration.ConversionListener.class.getDeclaredMethod("getFlag",String.class);
+        getFlagMethod.setAccessible(true);
+
+        boolean resBoolean = (Boolean) getFlagMethod.invoke(conversionListener,key);
+
+        Assert.assertTrue(resBoolean);
+
+        reset(analytics,app,context,sharedPreferences,conversionListener);
+    }
+
+    @Test
+    public void testAppsflyerIntegration_ConversionListener_setFlag_happyFlow() throws Exception {
+        String key="key";
+        boolean value=true;
+        Analytics analytics = mock(Analytics.class);
+        Application app = mock(Application.class);
+        Context context = mock(Context.class);
+        SharedPreferences sharedPreferences = mock(SharedPreferences.class);
+        SharedPreferences.Editor editor = mock(SharedPreferences.Editor.class);
+        when(analytics.getApplication()).thenReturn(app);
+        when(app.getApplicationContext()).thenReturn(context);
+        when(context.getSharedPreferences("appsflyer-segment-data",0)).thenReturn(sharedPreferences);
+        when(sharedPreferences.edit()).thenReturn(editor);
+        AppsflyerIntegration.ConversionListener conversionListener = spy(new AppsflyerIntegration.ConversionListener(analytics));
+        Method setFlagMethod = AppsflyerIntegration.ConversionListener.class.getDeclaredMethod("setFlag",String.class,boolean.class);
+        setFlagMethod.setAccessible(true);
+
+        setFlagMethod.invoke(conversionListener,key,value);
+
+        if(android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.GINGERBREAD){
+            verify(editor,times(1)).apply();
+        }
+        else{
+            verify(editor,times(1)).commit();
+        }
```

Comment:
The first condition is always true in our case.

Reply:
This code is from the AppsflyerIntegration class(I just want to check the correct method is called in this case), So if it is redundant here, is it redundant there?

Reply:
Yes, our min OS version is way above GINGERBREAD